File: hackerearth/codemonk/chandu-and-his-interns/chandu-and-his-interns.py
def sieve_ish(n):
    ''' too slow with CPython make PYPY available! >:( '''
    divisors = [1] * (n + 2)
    for i in range(2, n + 2):
        if divisors[i] < 4:
            for j in range(i, n + 2, i):
                divisors[j] += 1
    return divisors

def solve(xs):
    divisors = sieve_ish(max(xs))
    return '\n'.join('NO' if divisors[x] < 4 else 'YES' for x in xs)

if __name__ == '__main__':
    print solve([int(raw_input()) for _ in range(int(raw_input()))])

File: hackerearth/codemonk/chandu-and-his-interns/chandu-and-his-interns.c
#include <stdio.h>

int divisors[10000002];

void sieve_ish(int n, int divisors[]) {
  for (int i = 1; i < n; i++) {
    if (divisors[i] < 4) {
      for (int j = i; j < n; j += i) {divisors[j]++;}
    }
  }
}

int main() {
  int t, n;
  sieve_ish(10000002, divisors);
  scanf("%d", &t);
  for (int i = 0; i < t; i++) {
    scanf("%d", &n);
    if (divisors[n] < 4) {printf("NO\n");}
    else {printf("YES\n");}
  }
  return 0;
}
